$NetBSD: patch-meson.build,v 1.5 2025/06/09 19:12:35 vins Exp $

* Prevent reference to build directory in perl scripts.
* Fix detection of readline and asciidoctor.
* Fix iconv prototype mismatch on SunOS.

--- meson.build.orig	2025-06-01 13:59:03.000000000 +0000
+++ meson.build
@@ -89,7 +89,11 @@ conf.set_quoted('LOCALEDIR', prefix / ge
 conf.set_quoted('FVWM_CONFDIR', prefix / get_option('sysconfdir'))
 conf.set('RETSIGTYPE', 'void')
 conf.set('SIGNAL_RETURN', 'return')
-conf.set('ICONV_ARG_CONST', '')
+if host_machine.system() == 'sunos'
+    conf.set('ICONV_ARG_CONST', 'const')
+else
+    conf.set('ICONV_ARG_CONST', '')
+endif
 conf.set('fd_set_size_t', 'int')
 conf.set('EXECUTABLE_EXTENSION', 'NULL')
 
@@ -212,7 +216,7 @@ else
 endif
 
 # Perl is a required dependency, we generate shebangs at buildtime
-perl = find_program('perl', required: true)
+perl = find_program(get_option('perl'), required: true)
 
 golang = find_program('go', required: get_option('golang'))
 if golang.found()
@@ -226,19 +230,30 @@ if golang.found()
     endif
 else
     # the readline dependency is only required for FvwmConsole so let's only check for it if we're not building FvwmPrompt
-    readline = dependency('readline', required: get_option('readline'))
-    if readline.found()
-        all_found_deps += readline
-        conf.set10('HAVE_READLINE', true)
-        # Check for append_history to determine if we have GNU readline
-        if cc.has_function('append_history', dependencies: readline)
-            conf.set10('HAVE_GNU_READLINE', true)
+    readline = cc.find_library(
+		'edit', 
+		dirs: get_option('readline_lib'), 
+		has_headers: 'readline/readline.h',
+		required: false)
+    if readline.found() == false
+	readline = cc.find_library(
+		    'readline', 
+		    dirs: get_option('readline_lib'),
+		    has_headers: 'readline/readline.h', 
+		    required: get_option('readline'))
+	if readline.found()
+	    all_found_deps += readline
+	    conf.set10('HAVE_READLINE', true)
+	    # Check for append_history to determine if we have GNU readline
+            if cc.has_function('append_history', dependencies: readline)
+		conf.set10('HAVE_GNU_READLINE', true)
+	    endif
         endif
     endif
 endif
 
 # Sed is used for substitution in several configure_file calls
-sed = find_program('sed', required: true)
+sed = find_program('@SED@', required: true)
 
 # Optional dependencies
 # Tip: 'Feature' types always report 'not found' if disabled
@@ -403,7 +418,7 @@ endforeach
 build_docs = false
 if get_option('htmldoc') or get_option('mandoc')
     build_docs = true
-    asciidoc = find_program('asciidoctor', required: build_docs)
+    asciidoc = find_program('@ASCIIDOC@', required: build_docs)
     grep = find_program('grep')
     subdir('doc')
 endif
